health/manager.go

```diff
@@ -0,0 +1,303 @@
+/*
+Package health implements a simple tool for health data + metrics collection.
+
+We defines three main parts of the health monitorin tool: manager, collector and writer.
+- Writer description is avilable in its own package.
+- Collector is stored in collector.go file. It provides a simple interface that allows you to collect
+different type of health data per target. All methods should have targetID as a parameter. Collector
+will automatically send data to health manager.
+- Manger is a heart of the health collection tool. It initialize all comunication channels and has
+a control over collector and writer. Manager keeps all data collected by collector, calculates every
+target health once per cycle and sends calculated data to writer.
+*/
+package health
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	logging "github.com/zenoss/zenoss-go-sdk/health/log"
+	"github.com/zenoss/zenoss-go-sdk/health/target"
+	w "github.com/zenoss/zenoss-go-sdk/health/writer"
+	"github.com/zenoss/zenoss-go-sdk/utils"
+)
+
+// Manager keeps all information about targets and provides a functionality
+// to initialize collector, writer and communication channels
+type Manager interface {
+	// Start method should initialize collector with it's configuration and
+	// define a method to send data to a writer
+	Start(ctx context.Context)
+	// AddTargets provides a simple interface to register monitored targets
+	AddTargets(targets []*target.Target)
+}
+
+// NewManager creates a new Manager instance with internal target registry.
+// You should init configuration and writer before and pass it here
+func NewManager(ctx context.Context, config *Config, writer w.HealthWriter) Manager {
+	healthReg := newHealthRegistry()
+
+	return &healthManager{
+		registry: healthReg,
+		config:   config,
+		writer:   writer,
```

Comment:
Support for more than one destination? We may have Cloud + log enabled in ZDM, for example.

Reply:
Multiple destinations support added: writer handles a list of destinations